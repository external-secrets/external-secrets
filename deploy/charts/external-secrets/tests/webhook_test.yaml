suite: test webhook deployment
templates:
  - webhook-deployment.yaml
tests:
  - it: should match snapshot of default values
    asserts:
      - matchSnapshot: {}
  - it: should set imagePullPolicy to Always
    set:
      webhook.image.pullPolicy: Always
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: Always
  - it: should imagePullPolicy to be default value IfNotPresent
    asserts:
      - equal:
          path: spec.template.spec.containers[0].imagePullPolicy
          value: IfNotPresent
  - it: should override securityContext
    set:
      webhook.podSecurityContext:
        runAsUser: 2000
      webhook.securityContext:
        runAsUser: 3000
    asserts:
      - equal:
          path: spec.template.spec.securityContext
          value:
            runAsUser: 2000
      - equal:
          path: spec.template.spec.containers[0].securityContext
          value:
            allowPrivilegeEscalation: false
            capabilities:
              drop:
                - ALL
            readOnlyRootFilesystem: true
            runAsNonRoot: true
            runAsUser: 3000
            seccompProfile:
              type: RuntimeDefault
  - it: should override hostNetwork
    set:
      webhook.hostNetwork: true
    asserts:
      - equal:
          path: spec.template.spec.hostNetwork
          value: true
