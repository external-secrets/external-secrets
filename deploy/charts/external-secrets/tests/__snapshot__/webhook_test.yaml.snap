should match snapshot of default values:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: external-secrets-webhook
        app.kubernetes.io/version: v0.8.3
        helm.sh/chart: external-secrets-0.8.3
      name: RELEASE-NAME-external-secrets-webhook
      namespace: NAMESPACE
    spec:
      replicas: 1
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: external-secrets-webhook
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: external-secrets-webhook
        spec:
          automountServiceAccountToken: true
          containers:
            - args:
                - webhook
                - --port=10250
                - --dns-name=RELEASE-NAME-external-secrets-webhook.NAMESPACE.svc
                - --cert-dir=/tmp/certs
                - --check-interval=5m
                - --metrics-addr=:8080
                - --healthz-addr=:8081
              image: ghcr.io/external-secrets/external-secrets:v0.8.3
              imagePullPolicy: IfNotPresent
              name: webhook
              ports:
                - containerPort: 8080
                  name: metrics
                  protocol: TCP
                - containerPort: 10250
                  name: webhook
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 20
                periodSeconds: 5
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                readOnlyRootFilesystem: true
                runAsNonRoot: true
                runAsUser: 1000
                seccompProfile:
                  type: RuntimeDefault
              volumeMounts:
                - mountPath: /tmp/certs
                  name: certs
                  readOnly: true
          hostNetwork: false
          serviceAccountName: external-secrets-webhook
          volumes:
            - name: certs
              secret:
                secretName: RELEASE-NAME-external-secrets-webhook
